crates/compression/src/ports.rs

```diff
@@ -141,6 +141,28 @@ pub mod fault_proving {
             D::registry_root(self)
         }
     }
+
+    pub trait CompressedBlockRootProvider {
+        type Error: core::fmt::Display;
```

Comment:
Nit: I think it's more idiomatic to require `Error` here, which implies `Display`.
```suggestion
        type Error: core::error::Error;
```